### DISCUSSION TITLE
Linearized Stress Values from SCLs

### DISCUSSION THREAD

**Discussion by anthony-walker** [ansys/pymechanical#864] [Q&A]
Is it possible to reference SCLs by name and extract linearized stresses from these points for all along the SCL and all times?

**Comment by RobPasMue** [comment]
Hi @anthony-walker! To better redirect your question, could you please provide some context on the Ansys product that you are using? I believe it would be Ansys Mechanical, but I want to confirm first.

**Reply by anthony-walker** [reply]
@RobPasMue It is ansys mechanical.

**Comment by RobPasMue** [comment]
@ansys/pymechanical-maintainers could you please support this user?

**Comment by koubaa** [comment]
It is possible. This feature is called "Linearized Stress" in Ansys Mechanical. Apparently there is a free [addon](https://catalog.ansys.com/product/5b3bc6857a2f9a5c90d32ea2/pressure-vessels-a) that will automatically do this for pressure vessels, but I don't know exactly what it does.

**Comment by anthony-walker** [comment]
@koubaa I appreciate the response. I am looking to do this through pyansys using the results file. I want to reference already created SCLs and extract the linearized stresses programmatically. I don't believe the suggested addon will work for that.

**Reply by koubaa** [reply]
@anthony-walker For advanced post-processing with the MAPDL result file I would suggest looking at [pydpf](https://dpf.docs.pyansys.com/version/stable/).

**Comment by anthony-walker** [comment]
@koubaa I have been using `pydpf` but I haven't found a great way to reference already created. SCLs I figured this discussion may help lead me too that.

**Reply by koubaa** [reply]
@anthony-walker Can you be more specific about where the SCLs were created? In the mechanical application as Path objects?

**Reply by anthony-walker** [reply]
@koubaa yes, in mechanical as paths via the construction geometry.

**Reply by koubaa** [reply]
@anthony-walker I see. And you want to do compute the post-processing outside of Mechanical? Is that right?

**Reply by anthony-walker** [reply]
@koubaa that is correct. I want to access the stress values along the SCL and post process them outside of ansys

**Reply by koubaa** [reply]
@anthony-walker I see. The Path object defines the points at which the stress values are computed, but a Linearized Stress Result object can be used inside mechanical to actually compute those values, which displays them in a table. I believe you could access the linearized stress results along that path from python. Does that sound like what you need?

I initially thought you just wanted to get the points along a path and compute them yourself using something like pydpf.

**Reply by anthony-walker** [reply]
That is what I need. The values that were computed along the specified path(s). Ideally I want to reference these paths by their name or some identifier

**Reply by koubaa** [reply]
Hi Anthony, sorry for the delay. In the Mechanical UI, a linearized stress must select a path using a dropdown by name. The same can be done with scripting. Unfortunately, there isn't a direct script API to get the values of the linearized stress (these are presented in the tabular data grid in the UI), but a workaround does exist. You can sue `result.ExportToTextFile("/path/to/file.csv")` and parse the csv file for whatever information you need.

**Reply by anthony-walker** [reply]
@koubaa I am not seeing an `ExportToTextFile` method, which pyansys module is this in?

**Reply by koubaa** [reply]
@anthony-walker it's part of [Mechanical Scripting](https://mechanical.docs.pyansys.com/version/stable/user_guide_scripting/index.html), which PyMechanical gives access to.

**Reply by anthony-walker** [reply]
@koubaa can you provide a bit more information or point me to some examples/docs? The documentation I have come across seems to be lacking in describing how to connect to an existing mechanical session and get access to the `result` object your reference.

**Reply by anthony-walker** [reply]
@koubaa This also doesn't seem to be easily manipulated to capture data at multiple load/substeps

**Comment by dipinknair** [comment]
@anthony-walker https://catalog.ansys.com/product/5eb9b19c393ff65f64e331bc/linearized-stresse
let me know if this helps

**Comment by dipinknair** [comment]
@anthony-walker Hi, just checking in — any updates on this?
If there’s no response, we’ll plan to close this discussion 7 days due to inactivity.
Feel free to comment or reopen if needed.
